ArduCopter/crash_check.cpp

```diff
@@ -53,6 +58,62 @@ void Copter::crash_check()
     }
 }
 
+// check for loss of thrust and trigger thrust boost in motors library
+void Copter::thrust_loss_check()
+{
+    static uint16_t thrust_loss_counter;  // number of iterations vehicle may have been crashed
+
+    // exit immediately if motor failure handling already engaged
+    if (motors->get_thrust_boost()) {
+        return;
+    }
+
+    // return immediately if disarmed
+    if (!motors->armed() || ap.land_complete) {
+        thrust_loss_counter = 0;
+        return;
+    }
+
+    // check for desired angle over 15 degrees
+    // todo: add thrust angle to AC_AttitudeControl
+    const Vector3f angle_target = attitude_control->get_att_target_euler_cd();
+    if (safe_sqrt(sq(angle_target.x) + sq(angle_target.y)) > THRUST_LOSS_CHECK_ANGLE_DEVIATION_CD) {
```

Comment:
I don't understand why we're zeroing the counter for this condition

Reply:
is the condition backwards?

Reply:
This ensures we are descending despite being at maximum throttle and at low thrust angles. We expect to descend when at high bank or pitch angles.